pathways/inspections.py

```diff
@@ -509,19 +509,32 @@ def inspect(config, shipment, n_units_to_inspect):
         detected = False
         ret.boxes_opened_completion = n_units_to_inspect
         ret.stems_inspected_completion = n_units_to_inspect * inspect_per_box
-        for box in indexes_to_inspect:
+        for index in indexes_to_inspect:
             if not detected:
                 ret.boxes_opened_detection += 1
-            # In each box, loop through first n stems (n = inspect_per_box)
-            for stem in (shipment.boxes[box]).stems[0:inspect_per_box]:
+            # If inspecting full box, use box object for inspection
+            if within_box_pct in [1, 1.0]:
                 if not detected:
-                    ret.stems_inspected_detection += 1
-                if stem:
-                    # Count every infested stem in sample
-                    ret.infested_stems_completion += 1
-                    # If first infested box inspected, count all infested stems in box
+                    ret.stems_inspected_detection += stems_per_box
```

Comment:
Something like box.num_stems would fit with the following lines, base the number on the actual box, and remove the only use of stems_per_box in the `if within_box_pct in [1, 1.0]` branch.

```suggestion
                    ret.stems_inspected_detection += shipment.boxes[index].num_stems:
```